build/patches/experimental-user-scripts-support.patch

```diff
@@ -0,0 +1,10299 @@
+   ]
+ }
++
++chrome_java_sources += userscripts_java_sources
+diff --git a/chrome/android/proguard/main.flags b/chrome/android/proguard/main.flags
+--- a/chrome/android/proguard/main.flags
++++ b/chrome/android/proguard/main.flags
+@@ -63,3 +63,10 @@
+ -keep,allowobfuscation public class ** extends org.chromium.chrome.browser.base.SplitCompat*$Impl {
+   public <init>();
+ }
++
++# Keep fields name for all R.xml
++# used in SettingsUtils.java
++# components/browser_ui/settings/android/java/src/org/chromium/components/browser_ui/settings
++-keepclassmembers class **.R$xml* {
++    public static <fields>;
++}
+diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
+--- a/chrome/browser/BUILD.gn
++++ b/chrome/browser/BUILD.gn
+@@ -3371,6 +3371,11 @@ static_library("browser") {
+       ]
+       deps += [ "//chrome/android/modules/dev_ui/provider:native" ]
+     }
++    deps += [
++      "//components/user_scripts/common",
++      "//components/user_scripts/browser",
++      "//components/user_scripts/android",
++    ]
+   } else {
+     #!is_android
+     sources += [
+diff --git a/chrome/browser/about_flags.cc b/chrome/browser/about_flags.cc
+--- a/chrome/browser/about_flags.cc
++++ b/chrome/browser/about_flags.cc
+@@ -138,6 +138,7 @@
+ #include "components/translate/core/browser/translate_ranker_impl.h"
+ #include "components/translate/core/common/translate_util.h"
+ #include "components/ui_devtools/switches.h"
++#include "components/user_scripts/common/user_scripts_features.h"
+ #include "components/version_info/version_info.h"
+ #include "components/viz/common/features.h"
+ #include "components/viz/common/switches.h"
+@@ -6390,6 +6391,14 @@ const FeatureEntry kFeatureEntries[] = {
+      FEATURE_VALUE_TYPE(features::kUserDataSnapshot)},
+ #endif
+ 
++    {"enable-userscripts", flag_descriptions::kEnableUserScriptsName,
++     flag_descriptions::kEnableUserScriptsDescription, kOsDesktop | kOsAndroid,
++     FEATURE_VALUE_TYPE(user_scripts::features::kEnableUserScripts)},
++
++    {"enable-userscripts-log", flag_descriptions::kEnableLoggingUserScriptsName,
++     flag_descriptions::kEnableLoggingUserScriptsDescription, kOsDesktop | kOsAndroid,
++     FEATURE_VALUE_TYPE(user_scripts::features::kEnableLoggingUserScripts)},
++
+ #if defined(OS_WIN)
+     {"run-video-capture-service-in-browser",
+      flag_descriptions::kRunVideoCaptureServiceInBrowserProcessName,
+diff --git a/chrome/browser/flag_descriptions.cc b/chrome/browser/flag_descriptions.cc
+--- a/chrome/browser/flag_descriptions.cc
++++ b/chrome/browser/flag_descriptions.cc
+@@ -4899,6 +4899,16 @@ const char kUserDataSnapshotDescription[] =
+     "update and restoring them after a version rollback.";
+ #endif  // !defined(OS_ANDROID) && !BUILDFLAG(IS_CHROMEOS_ASH)
+ 
++const char kEnableUserScriptsName[] = "Enable user scripts";
++const char kEnableUserScriptsDescription[] =
++    "Enables experimental support on user scripts. "
++    "Please report back any bug.";
```

Comment:
This is implicit, no need to mention it.

Reply:
removed in https://github.com/bromite/bromite/pull/857/commits/8c164138391082350f20493a0ca5c4926887372c
no more need that flag